packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -250,7 +282,28 @@ function WorklistDataGrid(props) {
   useEffect(() => {
     setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [columns]);
+  }, [pinnedColumns, overflowColumns]);
+
+  // useEffect to calculate pinned column offsets
+  useEffect(() => {
+    if (pinnedColumns.length > 0) {
+      let cumulativeOffset = 0;
+      const offsetArray = [0];
```

Comment:
What is the advantage of initializing the first element to 0?

Reply:
This is because the left offset for the first column will always be 0, and this is an array of cumulative offsets. Also updated to use `cumulativeOffset` for the `offsetArray` initialization for more clarity: [80f7216](https://github.com/cerner/terra-framework/pull/1710/commits/80f72169793d55a4e2851b8b12d0051686391248)